### DISCUSSION TITLE
No Web Origin for SAML

### DISCUSSION THREAD

**Discussion by schurma** [keycloak/keycloak#12115] [Q&A]
I can't redirect from my backend-server (NodeJS) to our Keycloak SAML 2.0 Identity Provider (IdP).  Each are on separate IPs/Hosts.  Everything works great when I manually navigate to my backend's login page.  I'm redirected to the IdP for sign in and then my SAML assertion is posted back to my backend.  It's beautiful to see.  

But we can't use manual navigation and when our ajax call hits my backend login page and is redirected to the Keycloak IdP, the redirect fails with the preflight error, "No Access-Control-Allow-Origin header present on requested resource."  If we were using the Keycloak Open ID Connect (OIDC) IdP then I could use the "Web Origins" setting to solve this issue but we can't use OIDC and there is no "Web Origins" setting for the Keycloak SAML IdP.  

How do we enable "Web Origins" for Keycloak SAML?  How can I add my backend server's IP/Hostname to Keycloak SAML CORS or otherwise send the Access-Control-Allow-Origin I need?

**Comment by tacerus** [comment]
I'd be interested in this as well.

**Comment by TomKrcmar** [comment]
I'm having the exact same issue in a React app with an express backend.

I'm considering putting Keycloak behind an API Gateway that adds the CORS headers as a potential workaround, but I was not looking to set up this much infrastructure just for 4 headers.

**Comment by gbouille** [comment]
Hi all,
 A gentle up to this topic.
 I have exactly the same issue trying to use SAML with SOGo. Everything work like a charm except that.
 SOGo try an GET request that is reject by the browser with "No Access-Control-Allow-Origin header present on requested resource.". Then it try an OPTION request but receive an "Not allowed" response.

Thank you for considering this help request.
